tendrl/commons/objects/node/atoms/check_node_up/__init__.py

```diff
@@ -1,40 +1,81 @@
 from tendrl.commons.utils import log_utils as logger
 
 
 class CheckNodeUp(BaseAtom):
+    def __init__(self, *args, **kwargs):
+        super(CheckNodeUp, self).__init__(*args, **kwargs)
 
     def run(self):
-        fqdn = self.parameters.get("fqdn")
-        logger.log(
-            "info",
-            NS.publisher_id,
-            {"message": "Checking if node %s is up" %
-             self.parameters.get("fqdn")},
-            job_id=self.parameters['job_id'],
-            flow_id=self.parameters['flow_id']
-        )
-        out, err, rc = Command("ping -c 1 %s" % fqdn).run()
-        # and then check the response...
-        if not err and rc == 0:
-            logger.log(
-                "info",
-                NS.publisher_id,
-                {"message": "Ping to %s succeeded." %
-                 self.parameters.get("fqdn")},
-                job_id=self.parameters['job_id'],
-                flow_id=self.parameters['flow_id']
-            )
-            return True
-        else:
-            logger.log(
-                "info",
-                NS.publisher_id,
-                {"message": "Failed to ping %s. Error %s" % (
-                    self.parameters.get("fqdn"),
-                    err)},
-                job_id=self.parameters['job_id'],
-                flow_id=self.parameters['flow_id']
+        try:
+            flag = True
+            # check job is parent or child
+            job = NS.tendrl.objects.Job(
+                job_id=self.parameters['job_id']
+            ).load()
+            if "parent" not in job.payload:
+                # Getting node_ids from parameters
+                node_ids = self.parameters.get(
+                    'Node[]',
+                    None
+                )
+                if not node_ids:
+                    # fetch node id using integration_id
+                    integration_id = self.parameters[
+                        'TendrlContext.integration_id'
+                    ]
+                    key = "indexes/tags/tendrl/integration/%s" % \
+                        integration_id
+                    node_ids_str = etcd_utils.read(key).value
+                    node_ids = json.loads(node_ids_str)
+                # identifying fqdn using node_id
+                hosts = []
+                for node in node_ids:
+                    key = "nodes/%s/NodeContext/fqdn" % node
+                    fqdn = etcd_utils.read(key).value
+                    hosts.append(str(fqdn))
+                logger.log(
+                    "info",
+                    NS.publisher_id,
+                    {"message": "Checking if nodes %s are up" % hosts},
+                    job_id=self.parameters['job_id'],
+                    flow_id=self.parameters['flow_id']
+                )
+                nodes_up = []
+                nodes_down = []
+                for fqdn in hosts:
+                    out, err, rc = Command("ping -c 1 %s" % fqdn).run()
```

Comment:
I should have been more clear, we dont need to ping the nodes, we only need to load their NodeContext and check if NodeContext.status  is not "down" for any nodes, fail import if any NodeContext.status is "down"